package/network/services/dropbear/Makefile

```diff
@@ -142,6 +144,12 @@ define Package/dropbear/install
 	$(INSTALL_DIR) $(1)/usr/lib/opkg/info
 	$(INSTALL_DIR) $(1)/etc/dropbear
 	touch $(1)/etc/dropbear/dropbear_rsa_host_key
+ifeq ($(CONFIG_DROPBEAR_ECDSA_KEYGEN),y)
+	sed -i \
+		'/[ -s \/etc\/dropbear\/dropbear_rsa_host_key ] || keygen rsa/a \\t[ -s \/etc\/dropbear\/dropbear_ecdsa_host_key ] || keygen ecdsa' \
+		$(1)/etc/init.d/dropbear
+	touch $(1)/etc/dropbear/dropbear_ecdsa_host_key
+endif
```

Comment:
this is way too ugly. can't you runtime detect what key should be generated ?

Reply:
The "dropbearkey -h" output could maybe used to get the supported keytypes.

I thought there are maybe users who have ECC support activated, but don't want to generate an ecdsa host key. So I deciced to add an new extra option. 
Autodetection would be difficult in this case.

I don't know if this is really a practical usecase and I also considered to simply add ecdsa keygen if DROPBEAR_ECC is activated.